homeassistant/components/vicare/sensor.py

```diff
@@ -221,6 +270,150 @@ class ViCareSensorEntityDescription(SensorEntityDescription, ViCareRequiredKeysM
         device_class=SensorDeviceClass.ENERGY,
         state_class=SensorStateClass.TOTAL_INCREASING,
     ),
+    ViCareSensorEntityDescription(
+        key=SENSOR_GAS_SUMMARY_CONSUMPTION_CURRENT_DAY,
+        name="Heating gas consumption current day",
+        native_unit_of_measurement=VOLUME_CUBIC_METERS,
+        value_getter=lambda api: api.getGasSummaryConsumptionHeatingCurrentDay(),
+        unit_getter=lambda api: api.getGasSummaryConsumptionHeatingUnit(),
```

Comment:
I'm looking at the code that uses the `unit_getter`. It modifies these entity description constants. That's not good since the constant is expected to not change. I suggest we refactor this in a separate PR before adding more descriptions.

If we can't return the correct Home Assistant unit directly from the lambda, we should probably pass the looked up unit separately to the entity instead of modifying the constant entity description.

Reply:
The thing here is that the user can change the unit on his heating device and the api will immediately deliver the value according to the new unit.
I don't really see how we can fix that. Doing the calculation on our own is not possible unfortunately.

Reply:
You should create a new key "default_unit_of_measurement" in the entity descriptions, and then use/update `attr_native_unit_of_measurement` instead.

Reply:
Ok, understood. Makes sense. I'll create a dedicated PR.

Reply:
please see https://github.com/home-assistant/core/pull/64782